northstar/src/runtime/minijail.rs

```diff
@@ -13,75 +13,50 @@
 //   limitations under the License.
 
 use super::{
+    config::Config,
     process::{
         exit_handle, waitpid, Error, ExitHandleWait, ExitStatus, Pid, ENV_NAME, ENV_VERSION,
     },
     state::Container,
-    OutputStream,
 };
-use crate::runtime::{Event, EventTx};
-use futures::channel::oneshot;
+use crate::runtime::{config::debug::Strace, EventTx};
+use bytes::{Buf, BytesMut};
 use itertools::Itertools;
 use log::{debug, error, info, trace, warn, Level};
 use nix::{
     fcntl::{self, fcntl, OFlag},
     sys::signal,
     unistd::{self, chown, pipe},
 };
-use npk::manifest::{Dev, Mount, MountFlag};
+use npk::manifest::{Dev, IoInput, IoOutput, Manifest, Mount, MountFlag};
 use std::{
     fmt, iter, ops,
     os::unix::prelude::RawFd,
     path::{Path, PathBuf},
     pin::Pin,
+    process::Stdio,
     task::{Context, Poll},
+    unimplemented,
 };
 use tokio::{
     fs,
-    io::{self, unix::AsyncFd, AsyncBufReadExt, AsyncRead, AsyncWriteExt, ReadBuf},
+    io::{self, unix::AsyncFd, AsyncBufReadExt, AsyncRead, AsyncWrite, AsyncWriteExt, ReadBuf},
+    process::Command,
     select, task, time,
 };
-
-// We need a Send + Sync version of minijail::Minijail
-struct MinijailHandle(::minijail::Minijail);
-
-unsafe impl Send for MinijailHandle {}
-unsafe impl Sync for MinijailHandle {}
-
-impl ops::Deref for MinijailHandle {
-    type Target = ::minijail::Minijail;
-
-    fn deref(&self) -> &Self::Target {
-        &self.0
-    }
-}
-
-impl ops::DerefMut for MinijailHandle {
-    fn deref_mut(&mut self) -> &mut Self::Target {
-        &mut self.0
-    }
-}
+use tokio_util::sync::CancellationToken;
 
 #[derive(Debug)]
-pub struct Minijail {
+pub struct Minijail<'a> {
     log_fd: i32,
     event_tx: EventTx,
-    run_dir: PathBuf,
-    data_dir: PathBuf,
-    uid: u32,
-    gid: u32,
+    config: &'a Config,
```

Comment:
is it worth dragging in lifetimes for the config?
in my experience, this will usually lead to more trouble down the road so I try to use clones where appropriate.

Reply:
In this case it's super easy with `'a` for `State` and all fields.